pyproject.toml

```
@@ -39,38 +64,125 @@ distance-dirty = "{next_version}.dev{distance}+d{build_date:%Y%m%d}"
 [tool.versioningit.write]
 file = "src/finddata/_version.py"
 
-[tool.versioningit.onbuild]
-source-file = "src/finddata/_version.py"
-build-file = "finddata/_version.py"
-replacement = "{next_version}"
-
-[tool.setuptools.packages.find]
-where = ["src"]
-exclude = ["tests*", "scripts*", "docs*", "notebooks*"]
-
-[tool.setuptools.package-data]
-"*" = ["*.yml","*.yaml","*.ini"]
-
-[project.scripts]
-finddata = "finddata.cli:main"
-
-[project.optional-dependencies]
-bashcompletion = ["argcomplete"]
-
+# -------------------- #
+# PyTest configuration #
+# -------------------- #
 [tool.pytest.ini_options]
-pythonpath = [
-  ".", "finddata", "scripts"
-]
+addopts = "-v --cov=finddata --cov-report=term-missing"
+pythonpath = [".", "src", "scripts", "finddata"]
 testpaths = ["tests"]
 python_files = ["test*.py"]
-norecursedirs = [".git", "tmp*", "_tmp*", "__pycache__", "*dataset*", "*data_set*"]
-markers = [
-    "mymarker: example markers goes here"
+norecursedirs = [
+  ".git",
+  "tmp*",
+  "_tmp*",
+  "__pycache__",
+  "*dataset*",
+  "*data_set*",
 ]
+markers = ["mymarker: example markers goes here"]
 
+# ------------------ #
+# Ruff configuration #
+# ------------------ #
 [tool.ruff]
+target-version = "py311"
 line-length = 120
-select = ["A", "ARG", "BLE", "E", "F", "I", "PT"]
-ignore = ["F403", "F405", # wild imports and  unknown names
+select = ["A", "ARG", "ASYNC", "BLE", "C90", "E", "F", "I", "N", "UP032", "W"]
+exclude = ["notebooks/**", "**/*.ipynb", "conftest.py"]
```

There are no notebooks so this would be better as
```suggestion
exclude = ["conftest.py"]
```